File: VB/DXSample/ViewModels/MainViewModel.vb
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.Docking

Namespace DXSample.ViewModels
	Public Class MainViewModel
		Inherits ViewModelBase

		Public Property Panels() As ObservableCollection(Of PanelViewModel)
			Get
				Return GetValue(Of ObservableCollection(Of PanelViewModel))()
			End Get
			Set(ByVal value As ObservableCollection(Of PanelViewModel))
				SetValue(value)
			End Set
		End Property

		Public Sub New()
			Panels = New ObservableCollection(Of PanelViewModel)() From {
				New PanelViewModel() With {
					.Caption = "One",
					.Content = "A regular panel",
					.TargetName = "documentGroup"
				},
				New PanelViewModel() With {
					.Caption = "Two",
					.Content = "A regular panel",
					.TargetName = "documentGroup"
				},
				New PanelViewModel() With {
					.Caption = "Three",
					.Content = "A regular panel",
					.TargetName = "documentGroup"
				},
				New PanelViewModel() With {
					.Caption = "Five",
					.Content = "A panel",
					.TargetName = "layoutGroup"
				}
			}
		End Sub
	End Class

	Public Class PanelViewModel
		Inherits ViewModelBase
		Implements IMVVMDockingProperties

		Public Property Caption() As String
			Get
				Return GetValue(Of String)()
			End Get
			Set(ByVal value As String)
				SetValue(value)
			End Set
		End Property
		Public Property Content() As String
			Get
				Return GetValue(Of String)()
			End Get
			Set(ByVal value As String)
				SetValue(value)
			End Set
		End Property
		Public Property TargetName() As String
			Get
				Return GetValue(Of String)()
			End Get
			Set(ByVal value As String)
				SetValue(value)
			End Set
		End Property
	End Class
End Namespace

File: CS/DXSample/ViewModels/MainViewModel.cs
using System.Collections.ObjectModel;
using DevExpress.Mvvm;
using DevExpress.Xpf.Docking;

namespace DXSample.ViewModels {
    public class MainViewModel : ViewModelBase {
        public ObservableCollection<PanelViewModel> Panels {
            get { return GetValue<ObservableCollection<PanelViewModel>>(); }
            set { SetValue(value); }
        }

        public MainViewModel() {
            Panels = new ObservableCollection<PanelViewModel>() {
                new PanelViewModel() { Caption = "One", Content = "A regular panel", TargetName = "documentGroup" },
                new PanelViewModel() { Caption = "Two", Content = "A regular panel", TargetName = "documentGroup" },
                new PanelViewModel() { Caption = "Three", Content = "A regular panel", TargetName = "documentGroup" },                
                new PanelViewModel() { Caption = "Five", Content = "A panel", TargetName = "layoutGroup" },
            };
        }
    }    

    public class PanelViewModel : ViewModelBase, IMVVMDockingProperties {
        public string Caption {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
        public string Content {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }        
        public string TargetName {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
    }
}